cmd/gen-go-sample/out.go

```diff
@@ -75,6 +75,21 @@ func (st *symTab) put(ident string, typ initType) error {
 	return nil
 }
 
+func (st *symTab) disambiguate(ident string, typ initType) string {
+	base := ident
+	sf := 1
+	_, ok := st.scope[ident]
+	for ok {
+		sf++
+		ident = fmt.Sprintf("%s%d", base, sf)
+		_, ok = st.scope[ident]
+	}
+	if err := st.put(ident, typ); err != nil {
+		panic("bad state: ident shouldn't have existed")
```

Comment:
agreed, returning `err` is preferred to panicking (even if the end result is the tool panicking at the top level)

Reply:
Sooooo, I was hesitating on this, because this error should never happen: it keeps looking up the name in the table (the `for` loop), and when the name does not exist, it puts the name back. I didn't like to return an `err`, because there shouldn't be an error from the user's perspective.

Is it OK to swallow the error here?

Reply:
I'm OK with swallowing the error as long as we comment as to why it's being swallowed.

Reply:
OK, sounds reasonable. +1 to comment